api/envoy/config/core/v3/protocol.proto

```diff
@@ -28,7 +28,35 @@ message TcpProtocolOptions {
       "envoy.api.v2.core.TcpProtocolOptions";
 }
 
+//
+message QuicKeepAliveSettings {
+  // The max interval in seconds for a connection to send keep-alive probing packets
+  // (with PING or PATH_RESPONSE).
+  //
+  // If :ref:`initial_interval_seconds <envoy_v3_api_field_config.core.v3.QuicKeepAliveSettings.initial_interval_seconds>` is zero, a client connection will use this value to
+  // start probing. In such case, the value should be smaller than
+  // :ref:`connection idle_timeout <envoy_v3_api_field_config.listener.v3.QuicProtocolOptions.idle_timeout>` to
+  // prevent idle timeout while not too small to throttle the connection or to flood the peer.
+  // If zero, disable keepalive probing.
+  google.protobuf.UInt32Value max_interval_seconds = 1
+      [(validate.rules).message = {required: true}];
+
+  // The interval in seconds to send the first 6 keep-alive probing packets to prevent
+  // connection from hitting the idle timeout. Subsequent probes will be sent, each
+  // one with an interval exponentially longer than previous one, till it reaches
+  // :ref:`max_interval_seconds <envoy_v3_api_field_config.core.v3.QuicKeepAliveSettings.max_interval_seconds>`. And the probes afterwards will always use :ref:`max_interval_seconds <envoy_v3_api_field_config.core.v3.QuicKeepAliveSettings.max_interval_seconds>`.
+  //
+  // The value should be smaller than :ref:`connection idle_timeout <envoy_v3_api_field_config.listener.v3.QuicProtocolOptions.idle_timeout>` to
+  // prevent idle timeout and smaller than max_interval to take effect.
+  //
+  // If zero, disable keepalive probing for a server connection. For a client
+  // connection, if :ref:`max_interval_seconds <envoy_v3_api_field_config.core.v3.QuicKeepAliveSettings.max_interval_seconds>` is also zero, do not keepalive, otherwise use max_interval to probe all the time.
+  google.protobuf.UInt32Value initial_interval_seconds = 2
```

Comment:
My understanding is that UINT32Value makes it a possible to distinguish between unset and zero. Since it is required, there is no possibility of unset.

I don't feel strongly about it; I defer to @envoyproxy/api-shepherds for these details.